src/tree/link.rs

```diff
@@ -61,21 +61,51 @@ impl<'a> LinkLocation<'a> {
         let mut link_str = link.as_ref();
 
         // Check for direct URLs or anchor links
-        if is_url(link_str) || link_str.starts_with('#') {
+        // TODO: parse local links into LinkLocation::Page
+        // Known bug: single "/" parsed into Url instead of Page
+        if is_url(link_str) || link_str.starts_with('#') || link_str.starts_with("/") {
             return LinkLocation::Url(link);
         }
 
-        // Check for local links starting with '/'
-        if link_str.starts_with('/') {
-            link_str = &link_str[1..];
-        }
+        // // Check for local links starting with '/'
+        // if link_str.starts_with('/') {
+        //     link_str = &link_str[1..];
+        // }
+
+        // Take only the first segment for page
+        link_str = link_str.split('#').collect::<Vec<&str>>()[0]
+            .split('/')
+            .collect::<Vec<&str>>()[0];
 
         match PageRef::parse(link_str) {
-            Err(_) => LinkLocation::Url(link),
+            Err(_) => LinkLocation::Url(Cow::Owned(link_str.to_owned())),
             Ok(page_ref) => LinkLocation::Page(page_ref.to_owned()),
         }
     }
 
+    pub fn parse_extra(link: Cow<'a, str>) -> Option<Cow<'a, str>> {
+        let link_str = link.as_ref();
+
+        // Check for direct URLs or anchor links
+        // Does not parse local links for now
+        if is_url(link_str) || link_str.starts_with('#') || link_str.starts_with('/') {
+            return None;
+        }
+
+        let mut split_anchor: Vec<&str> = link_str.splitn(2, "#").collect();
+        let mut split_path: Vec<&str> = split_anchor[0].splitn(2, "/").collect();
+        split_path[0] = "";
+        let mut path = split_path.join("/");
+        split_anchor[0] = &path;
+        path = split_anchor.join("#");
```

Comment:
For one of/both of these `/`, `#` splitting sections it might be nice to have a comment explaining the intent in words, possibly with a quick example or two.